### PR TITLE
Check for existing invitation during signup

### PR DESCRIPTION
**Pull request by joshuaj194**
Prevent a new user from signing up and accidentally creating a duplicate org when they have an existing invitation. Instead, show an error message that points them to check their email for an invite.

### CODE REVIEW

**Comment by joshuaj194** [comment]
@grrowe @neurosnap, the review on this needs to wait until the updates on Auth API are released (should happen tomorrow). @benjodo will update here once that is released

**Comment by joshuaj194** [comment]
To test:
- send an invite to add a new user
- instead of using the link in the email, go straight to the signup page and try to sign up using the email address you sent the invite to.

The form should not submit. Instead, you should see an error telling you to check your email for an invite from support@aptible.com

File: src/auth/signup.ts [hunk 1]
```diff
@@ -15,10 +15,35 @@ const log = createLog("signup");
 export const signup = thunks.create<CreateUserForm>(
   "signup",
   function* onSignup(ctx, next) {
-    const { company: orgName, name, email, password } = ctx.payload;
+    const {
+      company: orgName,
+      name,
+      email,
+      password,
+      challenge_token,
```
Comment (neurosnap) [inline src/auth/signup.ts:23]: This should be `camelCase`

**Review by neurosnap — COMMENTED** [review]
Overall looks good

**Review by joshuaj194 — COMMENTED** [review]
updated to use camelCase for `challengeToken`

**Review by grrowe — APPROVED** [review]
Lgtm, manually tested.

**Pull request merged**